crates/ln-dlc-node/src/lib.rs

```diff
@@ -105,6 +105,57 @@ pub enum PaymentFlow {
     Outbound,
 }
 
+#[derive(PartialEq, Debug, Clone)]
+pub enum ChannelState {
+    Pending,
+    Open,
+    Closed,
+    ForceClosedRemote,
+    ForceClosedLocal,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Channel {
+    pub id: Option<i32>,
+    // The `user_channel_id` is set by 10101 at the time the `Event::HTLCIntercepted` when
+    // we are attempting to create a JIT channel.
+    pub user_channel_id: String,
+    // Until the `Event::ChannelReady` we do not have a `channel_id`, which is derived from
+    // the funding transaction. We use the `user_channel_id` as identifier over the entirety
+    // of the channel lifecycle.
+    pub channel_id: Option<String>,
+    pub capacity: i64,
+    // Set at the `Event::ChannelReady`
+    pub funding_txid: Option<String>,
```

Comment:
🤔 I think we should use actual types for the model: `Uuid`, `ChannelId` (it's just a `[u8; 32]` so it's fine I guess), `Txid` and `u64`s for natural numbers.